lib/Dialect/Triton/IR/Ops.cpp

```diff
@@ -728,6 +729,22 @@ LogicalResult ReshapeOp::verify() {
 }
 
 //-- FpToFpOp --
+OpFoldResult FpToFpOp::fold(FoldAdaptor adaptor) {
+  auto srcVal = getSrc();
+  if (!isZeroConst(srcVal)) {
+    return {};
+  }
+
+  auto srcConstant = srcVal.getDefiningOp<arith::ConstantOp>();
+  assert(srcConstant && "Expected srcVal to be a zero constant");
+
+  auto dstTy = getType();
+  OpBuilder builder(srcConstant);
+  auto newConstant = builder.create<mlir::arith::ConstantOp>(
```

Comment:
Hmm, in general folders don't create new ops--they just return existing values or return some attributes. (The attribute will be turned into constant values via dialect's `materializeConstant` hook.